### DISCUSSION TITLE
Weekdays in different languages?

### DISCUSSION THREAD

**Discussion by Kaoala** [joeycastillo/Sensor-Watch#73] [Ideas]
Maybe its just a little too much extra, but I was woundering what you think about a Weekday display in different languages?
In my case I would be interested in a german layout but you could do it in all kind of languages (except you can Display it: https://joeycastillo.github.io/Sensor-Watch-Documentation/wig/display.html ).
For the german one it should be possible to display: MO, DI, MI, DO, FR, SA, SO.
I think doing a custom one just for me would also possible for me but maybe there should be an option to change it for every one?
BTW Im very limited in my  programming so that i cant show you an idea how you could do it

**Comment by tahnok** [comment]
Huh, hadn't even thought about localization, but it should be technically
possible.

I wonder if there are embedded C frameworks for this sort of thing?

**Reply by xurizaemon** [reply]
Here is a PR from @tahnok on this: https://github.com/joeycastillo/Sensor-Watch/pull/323

And some notes on a single language's change at https://chris.bur.gs/sensor-watch-r%C4%81-m%C4%81ori/ - here's an example change from my custom build: https://github.com/xurizaemon/Sensor-Watch/commit/0f1e6e9588859209e40f5df5503c0c4ff143004d

I plan to switch to @tahnok's PR on my next watch update 😄

**Comment by 0xBEEFDAAD** [comment]
Some stock allow localized weekdays,  e.g. the GW-M5610U has weekdays in six languages even though the weekday display is also in 7 segments.
The question is: changing the weekdays at compiletime (1) or allow language change in the clock settings (2)?
(1) would be a 1-line-change in the function watch_utility_get_weekday in watch_utility.c (possibly in a #ifdef ENGLISH .. #elif FRENCH ..
where the language-defines might be set in movement_config.h, just like the SIGNAL_TUNE_..
(2) would require a addition to movement_settings_t and the setting_face 
  * LA - weekday language
  *      Sets the language of the weekdays
and I propose a additional function watch_utility_get_weekday_localized with the settings as a additional parameter

I'm willing to investigate, but would like to
 - hear opinions about the preferred solution
 - get proposals for languages and the translated weekdays

The next step would be the localization of other texts (ALarm Every Day WEekends..) but I think this causes too much confusion if one set the wrong language..
